File: backend/app/Models/Appointment.php
<?php

namespace App\Models;

use App\Enums\AppointmentStatus;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Appointment extends Model
{
    protected $fillable = [
        'user_id',
        'scheduled_at',
        'status',
        'appointment_type',
        'doctor_name',
        'completed_at',
        'notes',
    ];

    protected function casts(): array
    {
        return [
            'status' => AppointmentStatus::class,
            'scheduled_at' => 'datetime',
            'completed_at' => 'datetime',
        ];
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function feedbacks(): HasMany
    {
        return $this->hasMany(Feedback::class);
    }
}

File: android/app/src/main/java/com/example/opticalsystem/data/model/Appointment.kt
package com.example.opticalsystem.data.model

import com.google.gson.annotations.SerializedName

data class AppointmentListResponse(
    val data: List<Appointment>,
)

data class Appointment(
    val id: Int,
    @SerializedName("scheduled_at")
    val scheduledAt: String? = null,
    val status: String? = null,
    @SerializedName("appointment_type")
    val appointmentType: String? = null,
    @SerializedName("doctor_name")
    val doctorName: String? = null,
)
